src/Module.zig

```diff
@@ -983,8 +983,10 @@ pub const Scope = struct {
         tree: ast.Tree,
         /// Whether this is populated or not depends on `zir_loaded`.
         zir: Zir,
-        /// Package that this file is a part of, managed externally.
-        pkg: *Package,
+        /// Package that this file is a part of, managed externally. This is
+        /// `null` for files not part of any package such as those processed
+        /// by `zig fmt` or `zig ast-check`.
+        pkg: ?*Package,
```

Comment:
rather than making this optional, the places where it is being set to null/undefined need to create a valid package